sdk/storage/azure-storage-common/src/crypt.cpp

```diff
@@ -151,10 +151,8 @@ namespace Azure { namespace Storage {
 
     std::vector<uint8_t> Sha256(const std::vector<uint8_t>& data)
     {
-      if (data.size() > std::numeric_limits<ULONG>::max())
-      {
-        throw std::runtime_error("Data size is too big.");
-      }
+      AZURE_ASSERT_MSG(data.size() <= std::numeric_limits<ULONG>::max(), "Data size is too big.");
```

Comment:
This precondition validation is happening in an internal method, deep in the user call stack. It is fine to keep this here, but we should add the assert much earlier on, to "fail fast" that is as close to the actual end caller providing this data.

Where is this method called from?